Compile/src/Producers/PrefiringWeightProducer.cc

```diff
@@ -1,162 +1,338 @@
+ * with small modifications to match our setup                      *
+ ********************************************************************/
+
 #include "Excalibur/Compile/interface/Producers/PrefiringWeightProducer.h"
 
+#include "Kappa/DataFormats/interface/Kappa.h"
 
-#include <boost/algorithm/string.hpp>
+#include "TFile.h"
+#include "TF1.h"
 #include "TH2.h"
-#include <stdio.h>
-#include "Kappa/DataFormats/interface/Kappa.h"
 
 std::string PrefiringWeightProducer::GetProducerId() const { return "PrefiringWeightProducer"; }
 
 void PrefiringWeightProducer::Init(ZJetSettings const& settings)
 {
-    const std::string prefiringJetWeightFilePath = settings.GetPrefiringJetWeightFilename();
-    const std::string prefiringJetWeightHistName = settings.GetPrefiringJetWeightHistName();
-    const std::string prefiringPhotonWeightFilePath = settings.GetPrefiringPhotonWeightFilename();
-    const std::string prefiringPhotonWeightHistName = settings.GetPrefiringPhotonWeightHistName();
-
-    // open file containing prefireing jet weight map
-    TFile* prefiringJetWeightFile = new TFile(prefiringJetWeightFilePath.c_str(), "READ");
-    LOG(INFO) << "Prefiring jet weight file: " << prefiringJetWeightFilePath;
-    if(prefiringJetWeightFile->IsOpen())
-    {
-        LOG(DEBUG) << "Opened prefiring jet weight file successful";
-    }
-    else
-    {
-        LOG(ERROR) << "Could not open prefiring jet weight file!";    
-    }
-    // read in the weight maps
-    h_prefmap_jet = (TH2F*) prefiringJetWeightFile->Get(prefiringJetWeightHistName.c_str());
-    // detach ownership of object and close file
-    h_prefmap_jet->SetDirectory(0);
-    prefiringJetWeightFile->Close();
-
-
-     // open file containing prefireing photon weight map
-    TFile* prefiringPhotonWeightFile = new TFile(prefiringPhotonWeightFilePath.c_str(), "READ");
-    LOG(INFO) << "Prefiring photo weight file: " << prefiringPhotonWeightFilePath;
-    if(prefiringPhotonWeightFile->IsOpen())
-    {
-        LOG(DEBUG) << "Opened prefiring photon weight file successful";
+    const std::string dataeraEcal_ = settings.GetDataEraECAL();
+    const std::string dataeraMuon_ = settings.GetDataEraMuon();
+    const bool useEMpt_ = settings.GetUseJetEMPt();
+    const double prefiringRateSystUncEcal_ = settings.GetPrefiringRateSystematicUnctyECAL();
+    const double prefiringRateSystUncMuon_ = settings.GetPrefiringRateSystematicUnctyMuon();
+    const double jetMaxMuonFraction_ = settings.GetJetMaxMuonFraction();
+
+    missingInputEcal_ = false;
+    missingInputMuon_ = false;
+
+    const std::string prefiringL1MapsPath = settings.GetPrefiringL1MapsPath();
+    file_prefiringmaps_ = std::make_unique<TFile>(prefiringL1MapsPath.c_str(), "read");
+    if (file_prefiringmaps_ == nullptr) {
+        missingInputEcal_ = true;
+        LOG(ERROR) << this->GetProducerId() << ": "
+            << "File with maps not found. All prefiring weights set to 0.";
```

Comment:
Is this a useful scenario? 
Could this lead to a division by 0?
Maybe better Fatal?

Reply:
Default value is 1. When `missingInputEcal_ = ture;` the whole calculation is skipped and the weight is 1. So this Error message from the CMSSW code is misleading. Should be 1.

Reply:
ok, I suggested that :D
thanks

Reply:
Should we may add a comment on this?
To make it more clear for future generations :D

Reply:
Comment added and Warning updated.